nce/NCE-Pairwise-SM/main.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+import random
+import logging
+
+import torch
+from torchtext import data
+
+from args import get_args
+from trec_dataset import TrecDataset
+from evaluate import evaluate
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+ch = logging.StreamHandler()
+ch.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(levelname)s - %(message)s')
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
+args = get_args()
+config = args
+
+torch.manual_seed(args.seed)
+
+if not args.cuda:
+    args.gpu = -1
+if torch.cuda.is_available() and args.cuda:
+    logger.info("Note: You are using GPU for training")
+    torch.cuda.set_device(args.gpu)
+    torch.cuda.manual_seed(args.seed)
+if torch.cuda.is_available() and not args.cuda:
+    logger.info("Warning: You have Cuda but do not use it. You are using CPU for training")
+np.random.seed(args.seed)
+random.seed(args.seed)
+
+QID = data.Field(sequential=False)
+AID = data.Field(sequential=False)
+QUESTION = data.Field(batch_first=True)
+ANSWER = data.Field(batch_first=True)
+LABEL = data.Field(sequential=False)
+EXTERNAL = data.Field(sequential=False, tensor_type=torch.FloatTensor, batch_first=True, use_vocab=False,
+                      preprocessing=data.Pipeline(lambda x: x.split()),
+                      postprocessing=data.Pipeline(lambda x, train: [float(y) for y in x]))
```

Comment:
@Victor0118 with the latest torchtext version, you'll have to replace the code with following:
```python
EXTERNAL = data.Field(sequential=True, tensor_type=torch.FloatTensor, batch_first=True, use_vocab=False,
            postprocessing=data.Pipeline(lambda arr, _, train: [float(y) for y in arr]))
```